docs/semgrep-supply-chain/upgrade-guidance.md

```diff
@@ -0,0 +1,125 @@
+---
+slug: upgrade-guidance
+title: Upgrade guidance
+hide_title: true
+description: Know if a vulnerable package or dependency can be easily and reliably upgraded to a fixed version.
+tags:
+  - Semgrep Supply Chain
+  - Semgrep AppSec Platform
+---
+
+import ViewDetailsSsc from "/src/components/procedure/_view-details-ssc.md"
+ 
+# Upgrade guidance
+
+Know if you or your developers can safely and reliably update a vulnerable package or dependency to a fixed version. From there, choose to:
+
+- Have Semgrep open a pull request (PR)
+- Create a Jira ticket
+- Set the finding's triage status as **To fix**
+
+Semgrep's dependency upgrade guidance uses AI to analyze if a finding can be **safely upgraded** or if upgrading the package can cause **breaking changes**. Semgrep can also create a PR to upgrade the package.
+
+## Feature maturity
+
+This feature is in **private beta**. To join the beta, reach out to the [Semgrep support team](/support).
+
+## Supported languages and package managers
+
+- **Python** codebases with the following package managers:
+  - `pip`
+  - `pip-tools`
+  - `pipenv`
+  - Poetry
+
+Semgrep supports the following source code managers:
+
+- GitHub Cloud and GitHub Enterprise Server (self-hosted)
+  - This **includes** projects added to Semgrep through Semgrep Managed Scans 
+
+## Prerequisites
+
+- At least one repository that scans for dependencies through Semgrep Supply Chain. See [Scan third-party dependencies](/semgrep-supply-chain/getting-started).
+- Semgrep Assistant must be [enabled](/semgrep-assistant/getting-started).
+- Your **private** GitHub Semgrep app must have **Read and write** access on the **Contents** permission. Refer to the following section for steps.
+
+### Grant **Read and write** access
+
+To change your Semgrep app's permission:
+
+1. Navigate to the settings page of your private Semgrep GitHub app; refer to [<i class="fas fa-external-link fa-xs"></i> Changing the permissions of a GitHub app](https://docs.github.com/en/apps/maintaining-github-apps/modifying-a-github-app-registration#changing-the-permissions-of-a-github-app) for steps.
+1. In the **Repository permissions** section, search for `Contents`.
+1. Click the drop-down menu and select **Read and write**.
+
+## How it works
+
+After enabling dependency guidance, Semgrep performs post-scan analysis and marks applicable findings as **Safe to upgrade** or with **Breaking changes**.
+
+- Frequency: this analysis is performed every **two hours** on the latest **full scan**.
+- Only findings **with fixed versions** are marked by Semgrep as **Safe to upgrade** or with **Breaking changes**.
```

Comment:
What does fixed versions mean? Pinned versions?

Reply:
Pinned version: they're keeping it at that version and not upgrading it
Fixed version: version where the vulnerability has been fixed / patched

Will edit this to make it clear.